### PR TITLE
Adding loging module 

### PR DESCRIPTION
**Pull request by karthiknatarajan-27558**
- Added a log config 
- Added console logging 
- Added file logging for Error logs

### CODE REVIEW

**Comment by ndeoligence** [comment]
@karthiknatarajan-27558 thanks for the PR. I just have a concern here that I'd like to voice. But since I'm not an expert I'm hoping someone else will chime in.

I'm just worried about replacing all printouts with logging. This is fine on messages that aren't meant for a CLI  user, but it could hide important messages meant for such a user. This is because the logs might be redirected away from the user (to, say, a file. Or simply silenced via log levels) and they only need to view them to help debug the code (like when it's behaving strangely).

TLDR: But I think the user-directed (ie instead of dev-facing) messages should remain as print statements, rather than use logs.
Please correct me if I'm wrong here.
